code-review.js

```diff
@@ -1,60 +1,115 @@
+"use strict";
 /**
  * A priority queue stores a list of items but each can have a numeric priority value.
  * Items with a higher priority are dequeued before items with a lower priority.
  * Implemented as a hash of arrays where the hash keys are priority values.
+ *
+ * @class
  */
-function PriorityQueue(size) {
+function PriorityQueue() {
 	this.store = {};	// keys are priorities, values are arrays of elements
 	this.count = 0;
 
-	// adds an item
-	// priority must be an integer (higher value has higher priority)
+
+	/**
+	 * Adds an item
+	 * priority must be a positive integer (higher value has higher priority)
+	 * 
+	 * @param {*} value - The Item we wish to change
+	 * @param {Number} priority - Positive integer 
+	 */
 	this.add = function(value, priority) {
-		if (this.store[priority] == undefined)
+		if (this.store[priority] == undefined) {
 			this.store[priority] = [];
+		}
 
 		this.store[priority].push(value);
 		this.count++;
 	};
 
-	// returns the oldest-added value with the highest priority
-	this.Pop = function() {
-		maxKey = Math.max(Object.keys(this.store));
+	/**
+	 * Returns the oldest-added value with the highest priority
+	 */
+	this.pop = function() {
+		if (this.count === 0) {
+			return false;
+		}
+		var maxKey = Math.max.apply(null, this.getAllPriorities());

```

Comment:
`Math.max` does not accept array, needed apply call to get the highest priority.